### ISSUE TITLE
add ability to signal rclone to finish current partial task and exit

### ISSUE THREAD

**Issue by trapexit** [ncw/rclone#966]
I'd be convenient to be able to signal rclone (perhaps via a SIGUSR1 or SIGUSR2?) to finish up working on the current transfers and then exit so uploads aren't left in a half state. Or perhaps SIGINT? One ctrl-c on the terminal or from external source could tell it to finish up and exit while a second SIGINT would instruct it to exit immediately.

**Comment by ncw** [comment]
Nice idea.

**Comment by martinbowling** [comment]
+1

**Comment by marrie** [comment]
I agree. If you issue the "brake" command you could be presented with, "finish task or abort" this suggestion for those of us who use key strokes or what not to control our screen sessions.

Anothr thing, finish task could be keyed to f and abort could be keyed to a or something like that.

**Comment by kk7ds** [comment]
Yes please!

**Comment by felixbuenemann** [comment]
For interactive use I think the first Ctrl+C (SIGINT) for soft abort and second for hard abort works best. For non-interactive use one could probably just send SIGINT for soft abort and SIGTERM for hard abort.

**Comment by roberdaniel** [comment]
I adhere to this feature request.
Another approach could be to put some "flag" that could be read from time to time, for example if you put a file in the same folder where rclone.exe resides, named stop.txt, after X minutes Rclone could look up if there is the file and then to gracefully stop after end the current file being transfered.
Maybe is not the most elegant solution but it's handy.

**Comment by ncw** [comment]
This could be an API call too...

**Comment by DurvalMenezes** [comment]
This would be a great feature to have for some use cases here; I suggest implementing it via the remote control server (ie, `--rc`) functionality.

**Comment by ColinHeathman** [comment]
I was able to figure out a workaround using `socat` as a “session manager” for graceful shutdown. example:

`rclone serve sftp --addr :8022 local:/my_dir/` can be used to host the `/my_dir` folder with SFTP
`socat TCP-LISTEN:2022,fork,reuseaddr TCP-CONNECT:127.0.0.1:8022` was used to proxy connections to the real rclone instance. 

socat would start with a certain pid (eg. 150) when an sftp session was opened, socat would “move” the traffic proxy into a new process (eg. 151). Then a SIGTERM signal could be sent to the original process (150) and the container would stop accepting new connections. The child processes can then be monitored using something like ps and will stay open until the client disconnects. Once that happens, the rclone process can be safely terminated.

I will write a script to automate this and I can post it as a gist.
